samples/android/app/src/main/java/org/hyperledger/indy/sdk/ui/BaseActivity.kt

```diff
@@ -0,0 +1,211 @@
+package org.hyperledger.indy.sdk.ui
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.google.android.material.snackbar.Snackbar
+import kotlinx.android.synthetic.main.activity_base.view.*
+import kotlinx.android.synthetic.main.base_main_content.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.withContext
+import org.hyperledger.indy.sdk.R
+import org.hyperledger.indy.sdk.databinding.ActivityBaseBinding
+import org.hyperledger.indy.sdk.pool.Pool
+import org.hyperledger.indy.sdk.utils.PoolUtils
+import org.hyperledger.indy.sdk.wallet.Wallet
+import org.json.JSONObject
+
+abstract class BaseActivity : AppCompatActivity() {
+
+    // view bindings
+    private lateinit var baseBinding: ActivityBaseBinding
+    private lateinit var mainContentView: View
+    lateinit var job: Job
+
+    val issuerDid = "NcYxiDXkpYi6ov5FcYDi1e"
+    val proverDid = "VsKV7grR1BUE29mG2Fm2kX"
+
+    // issuer
+    val issuerWalletId = "issuerWallet"
+    val issuerWalletKey = "issuer_wallet_key"
+
+    // prover
+    //val proverWalletId = "trusteeWallet"
+    val proverWalletId = "proverWallet"
+    val proverWalletKey = "prover_wallet_key"
+
+    // me
+    val myWalletId = "myWallet"
+    val myWalletKey = "my_wallet_key"
+
+    // their
+    val theirWalletId = "theirWallet"
+    val theirWalletKey = "their_wallet_key"
```

Comment:
I think you don't need that yous pass values directly in function when needed
And also this doesn't belong into base Activity.